flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceFailover.java

```diff
@@ -110,6 +113,14 @@ protected Schema schema() {
     return TestFixtures.SCHEMA;
   }
 
+  protected HadoopTableExtension getSourceTableExtension() {
```

Comment:
why are these needed? we typically don't add such methods to tests. Also if this needs to be accessed, then the constant can be accessed directly as it has the same visibility as this method

Reply:
This is for the inherited class `TestIcebergSourceFailoverWithWatermarkExtractor`. Specifically this is for the tests in this class to use its own `SOURCE_TABLE_EXTENSION` and `SINK_TABLE_EXTENSION` constants. by default, the two constants in the original class `TestIcebergSourceFailover` are used in `TestIcebergSourceFailoverWithWatermarkExtractor` tests.

But I feel the same like your comment. Let me change this part to non-use of these methods.

Reply:
Pushed the new classes, but still working on these classes.